pkg/util/util.go

```diff
@@ -313,16 +313,23 @@ func WriteFileAtomic(filename string, data []byte, perm os.FileMode) error {
 	if err != nil {
 		return err
 	}
+
 	n, err := f.Write(data)
 	f.Close()
-	if err == nil && n < len(data) {
+	if err != nil {
+		return err
+	}
+
+	if n < len(data) {
```

Comment:
This can still be an unreachable branch. I'm glad to remove it if you ask for.

BTW, I have no idea why IT fails, it seems unrelated to this pr (fail to start tidb).

https://ci2.pingcap.net/blue/organizations/jenkins/binlog_ghpr_integration/detail/binlog_ghpr_integration/7/pipeline/